packages/atauth/src/auth_cli.c

```diff
@@ -490,9 +524,13 @@ int get_apkam_key(char **key, const char *key_name, atclient_connection *ctx, co
     return ret;
   }
   // extract the key from the json
-  cJSON *key_json = cJSON_GetObjectItemCaseSensitive(json_server_resp, "value");
+  const cJSON *key_json = cJSON_GetObjectItemCaseSensitive(json_server_resp, "value");
   if (cJSON_IsString(key_json) && key_json->valuestring != NULL) {
-    *key = strdup(key_json->valuestring);
+    strcpy(key, key_json->valuestring);
+  }
+  const cJSON *iv_json = cJSON_GetObjectItemCaseSensitive(json_server_resp, "iv");
+  if (cJSON_IsString(iv_json) && iv_json->valuestring != NULL) {
+    strcpy((char *)iv, iv_json->valuestring);
```

Comment:
I'm not entirely sure of what `cJSON_GetObjectItemCaseSensitive`, but just do a double check in case it is possible for this function to return NULL, and if so, do a NULL check